staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -230,7 +230,41 @@ func (h *UpgradeAwareHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 	proxy := httputil.NewSingleHostReverseProxy(&url.URL{Scheme: h.Location.Scheme, Host: h.Location.Host})
 	proxy.Transport = h.Transport
 	proxy.FlushInterval = h.FlushInterval
-	proxy.ServeHTTP(w, newReq)
+	proxy.ServeHTTP(newFlushHeaderWriter(w), newReq)
+}
+
+// TODO(liggitt): drop this once https://github.com/golang/go/issues/31125 is fixed
+func newFlushHeaderWriter(w http.ResponseWriter) http.ResponseWriter {
+	flusher, ok := w.(http.Flusher)
+	if !ok {
+		return w
+	}
+	hijacker, ok := w.(http.Hijacker)
```

Comment:
hijacker and closeNotifier are not used by the implementation. They should not be required to create the flushHeaderWriter

Reply:
they're used by the [reverse proxy implementation](https://golang.org/src/net/http/httputil/reverseproxy.go?s=488:2518#L17), so we can't mask those methods from the original writer

Reply:
My bad, I failed to notice the interfaces where actually embedded in the struct.
A small comment in the code explaining why they are needed would be awesome (because it is not really obvious)